### ISSUE TITLE
bugs in document_embedding

### ISSUE THREAD

**Issue by joonable** [PacktPublishing/Natural-Language-Processing-with-TensorFlow#2]
Hello. I found a bug while looking through the code, [ch4/ch4_document_embedding.ipynb ](https://github.com/PacktPublishing/Natural-Language-Processing-with-TensorFlow/blob/master/ch4/ch4_document_embedding.ipynb)

Batch is not just a vector, but a matrix. I added the code to **generate batches of data for skip-gram** 

`

    batch = np.ndarray(shape=(batch_size, span-1), dtype=np.int32)
    labels = np.ndarray(shape=(batch_size, 1), dtype=np.int32)

`

below this code, you can find a for-loop

`

    for i in range(batch_size // num_samples):
        k = 0
        for j in list(range(window_size)) + list(range(window_size + 1, 2 * window_size + 1)):
            batch[i * num_samples + k] = buffer[window_size]
            labels[i * num_samples + k, 0] = buffer[j]
            k += 1

`

In this for loop, batch is considered as an array(or a vector), not a matrix. Because of this problem, after the for-loop, batch contains same value when axis=0. For example,

`

array([[2, 2, 2, 2],
       [2, 2, 2, 2],
       [2, 2, 2, 2],
       [2, 2, 2, 2],
       [3, 3, 3, 3],
       [3, 3, 3, 3],
       [3, 3, 3, 3],
       [3, 3, 3, 3]], dtype=int32)

`  
I think these're not appropriate training data for doc2vec algorithm. Can you please check if it has a problem or just I misunderstand something. Thank you.
